commons-numbers-complex-arrays/src/test/java/org/apache/commons/numbers/complex/arrays/ComplexListTest.java

```diff
@@ -137,92 +256,119 @@ void testSetAddAndAddAllNullPointerException() {
 
-        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> list.get(-1));
-        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> list.get(size));
-        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> list.get(size + 1));
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> l.get(-1));
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> l.get(size));
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> l.get(size + 1));
 
-        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> list.set(-2, Complex.ofCartesian(200, 1)));
-        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> list.set(size, Complex.ofCartesian(200, 1)));
-        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> list.set(size + 1, Complex.ofCartesian(200, 1)));
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> l.set(-2, Complex.ofCartesian(200, 1)));
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> l.set(size, Complex.ofCartesian(200, 1)));
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> l.set(size + 1, Complex.ofCartesian(200, 1)));
     }
 
-    @Test
-    void testAddIndexOutOfBoundExceptions() {
-        ComplexList list = ComplexList.interleaved();
+    @ParameterizedTest
+    @MethodSource({"listImplementations"})
+    void testAddIndexOutOfBoundExceptions(List<Complex> l) {
         int size = 5;
-        IntStream.range(0, size).mapToObj(i -> Complex.ofCartesian(i, -i)).forEach(list::add);
+        IntStream.range(0, size).mapToObj(i -> Complex.ofCartesian(i, -i)).forEach(l::add);
 
         Assertions.assertThrows(IndexOutOfBoundsException.class, () ->
-            list.add(-1, Complex.ofCartesian(42, 13)));
+            l.add(-1, Complex.ofCartesian(42, 13)));
         Assertions.assertThrows(IndexOutOfBoundsException.class, () ->
-            list.add(size + 1, Complex.ofCartesian(42, 13)));
+            l.add(size + 1, Complex.ofCartesian(42, 13)));
     }
 
-    @Test
-    void testRemoveIndexOutOfBoundExceptions() {
-        ComplexList list = generateComplexList(2);
-        list.remove(0);
-        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> list.remove(1));
-        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> list.remove(-1));
+    @ParameterizedTest
+    @MethodSource({"listImplementationsWithSize"})
+    void testRemoveIndexOutOfBoundExceptions(List<Complex> l, int size) {
+        IntStream.range(0, size).forEach(i -> l.add(Complex.ofCartesian(i, -i)));
+        l.remove(0);
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> l.remove(1));
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> l.remove(-1));
     }
 
     @ParameterizedTest
     @ValueSource(ints = {0, 1, 10})
     void testConstructor(int size) {
```

Comment:
Note you should be able to stream a function reference as the argument to the parameterized test: IntToObjFunction<List<Complex>>:
```Java
    void testConstructor(IntFunction<List<Complex>> constructor, int size) {
        List<Complex> list = constructor.apply(size);
```
This requires:
```Java
Arguments.of((IntFunction<List<Complex>>) ComplexList::interleaved, 0),
Arguments.of((IntFunction<List<Complex>>) ComplexList::interleaved, 1), ...
```